### PR TITLE
Added All Your Base exercise

### PR DESCRIPTION
**Pull request by icyrockcom**
Let me know how it looks.

### CODE REVIEW

File: exercises/all-your-base/examples/src/AllYourBase.purs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+rebase' :: Int -> Int -> Array Int -> Array Int
+rebase' inb oub ind = toDec # fromDec # reverse
+  where toDec = zipWith (*) ind pows # sum
+        pows = map (pow inb) (range (length ind - 1) 0)
+        fromDec 0 = []
+        fromDec n = n `mod` oub : fromDec (n / oub)
+
+rebase :: Int -> Int -> Array Int -> Maybe (Array Int)
+rebase inb oub ind
+  | maybe true ((==) 0) (head ind) = Nothing
```
Comment (paf31) [inline exercises/all-your-base/examples/src/AllYourBase.purs]: You can use `any` here:

`any (_ == 0) (head ind)`

File: exercises/all-your-base/examples/src/AllYourBase.purs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  | any ((<=) inb) ind             = Nothing
```
Comment (paf31) [inline exercises/all-your-base/examples/src/AllYourBase.purs]: A section `inb <= _` would be clearer here.

File: exercises/all-your-base/examples/src/AllYourBase.purs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  | inb < 2 || oub < 2             = Nothing
+  | any ((>) 0) ind                = Nothing
```
Comment (paf31) [inline exercises/all-your-base/examples/src/AllYourBase.purs]: You could combine this with the other section:

```
any ((_ < 0) || (inb <= _)) ind
```

File: exercises/all-your-base/examples/src/AllYourBase.purs [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+import Data.Array (any, head, length, range, reverse, zipWith, (:))
+import Data.Foldable (sum)
+import Data.Int (pow)
+import Data.Maybe (Maybe(..), maybe)
+
+rebase' :: Int -> Int -> Array Int -> Array Int
+rebase' inb oub ind = toDec # fromDec # reverse
+  where toDec = zipWith (*) ind pows # sum
+        pows = map (pow inb) (range (length ind - 1) 0)
+        fromDec 0 = []
```
Comment (paf31) [inline exercises/all-your-base/examples/src/AllYourBase.purs]: Use `unfoldr` here.

File: exercises/all-your-base/examples/src/AllYourBase.purs [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+import Prelude
+import Data.Array (any, head, length, range, reverse, zipWith, (:))
+import Data.Foldable (sum)
+import Data.Int (pow)
+import Data.Maybe (Maybe(..), maybe)
+
+rebase' :: Int -> Int -> Array Int -> Array Int
+rebase' inb oub ind = toDec # fromDec # reverse
+  where toDec = zipWith (*) ind pows # sum
```
Comment (paf31) [inline exercises/all-your-base/examples/src/AllYourBase.purs]: I would inline `toDec`:

```purescript
rebase' inb oub ind = 
    zipWith (*) ind pows
    # sum
    # fromDec
    # reverse
  where
    ...
```

**Comment by paf31** [comment]
Thanks for the PR! Looks good, but I have a few refactoring ideas.

File: exercises/all-your-base/examples/src/AllYourBase.purs (hunk 1 above)
Comment (icyrockcom) [inline exercises/all-your-base/examples/src/AllYourBase.purs]: Makes sense, though I had to change to `all` to cover "empty list" case.

File: exercises/all-your-base/examples/src/AllYourBase.purs (hunk 5 above)
Comment (icyrockcom) [inline exercises/all-your-base/examples/src/AllYourBase.purs]: Sure thing, updated. Interesting, the original form seems more natural to me. :)

**Comment by icyrockcom** [comment]
Thanks @paf31 - updated these per the other comments.

**Comment by paf31** [comment]
Thanks, looks great!

**Pull request merged**